src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorComparer.cs

```diff
@@ -61,7 +62,11 @@ public virtual bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor
                     descriptorX.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     descriptorY.AllowedChildren.OrderBy(child => child, StringComparer.OrdinalIgnoreCase),
                     StringComparer.OrdinalIgnoreCase))) &&
-                descriptorX.TagStructure == descriptorY.TagStructure;
+                descriptorX.TagStructure == descriptorY.TagStructure &&
+                Enumerable.SequenceEqual(
+                    descriptorX.PropertyBag.OrderBy(propertyX => propertyX.Key, StringComparer.OrdinalIgnoreCase),
```

Comment:
The `PropertyBag` was constructed with a `StringComparer.Ordinal` why would this differ?